.buildkite/scripts/build_and_upload_firecracker_packages.sh

```diff
@@ -5,56 +5,32 @@ set -euo pipefail
 ################################################################################
 # This file is responsible for building and uploading non-Docker packages, like
 # the Firecracker microVM kernel and the Firecracker rootfs.
-# It will not upload anything if a file of that name and SHA already exist
-# upstream.
+#
+# It will not upload or promote anything if the manifest file for the artifact
+# specifies the same version currently in ${UPSTREAM_REGISTRY}.
+# (As such, it is convenient to include a checksum of the input files or
+# resulting artifact in the version!)
 ################################################################################
 
 source .buildkite/scripts/lib/artifacts.sh
-source .buildkite/scripts/lib/version.sh
-source firecracker/constants.sh
-
-DEFAULT_VERSION="$(timestamp_and_sha_version)"
-
-get_version() {
-    # Special case for Firecracker kernel - we just label it with the
-    # Firecracker release/kernel vm built
-
-    if [[ "${1}" == "firecracker_kernel.tar.gz" ]]; then
-        echo "firecracker-${FIRECRACKER_RELEASE}-kernel-${KERNEL_VERSION}"
-        return 0
-    fi
-    echo "${DEFAULT_VERSION}"
-}
-
-(
-    echo "--- Building packages"
-    make dist/firecracker_kernel.tar.gz
-)
-
-# These will be uploaded to Cludsmith with just their basename.
-readonly PACKAGES=(
-    dist/firecracker_kernel.tar.gz
-)
 
 # This is where our images will ultimately be promoted to. It is the
-# registry we'll need to query to see if file with the same contents
+# registry we'll need to query to see if file with the same version
 # already exists.
 readonly UPSTREAM_REGISTRY="${UPSTREAM_REGISTRY:-grapl/testing}"
 readonly UPLOAD_TO_REGISTRY="${UPLOAD_TO_REGISTRY:-grapl/raw}"
 
-# This is the list of packages that actually have different shas
-new_packages=()
-
-# Returns a string like `2de4f0c...`
-sha256_of_file() {
-    sha256sum "${1}" | awk '{print $1;}'
+get_version_from_manifest() {
+    local -r artifact_path="${1}"
+    local -r manifest_path="${artifact_path}.manifest"
+    jq -r ".version" "${manifest_path}"
```

Comment:
This whole function should be in the `manifest.sh` Bash library.

Ideally, nothing outside of that library file would need to be aware of the internal structure of the manifest file, making future evolution much easier.